Code/network/src/Client.cpp

```diff
@@ -0,0 +1,105 @@
+#include "Client.h"
+
+Client::Client(const Endpoint& acRemoteEndpoint)
+    : m_connection(*this, acRemoteEndpoint)
+    , m_socket(acRemoteEndpoint.GetType(), false)
+{
+    m_socket.Bind();
+}
+
+void Client::Disconnect() noexcept
+{
+    if (!m_connection.GetState() == Connection::kNone)
+    {
+        m_connection.Disconnect();
+    }
+}
+
+bool Client::Send(const Endpoint& acRemoteEndpoint, Buffer aBuffer) noexcept
+{
+    Socket::Packet packet{ acRemoteEndpoint, std::move(aBuffer) };
+    return m_socket.Send(packet);
+}
+
+bool Client::SendPayload(uint8_t *apData, size_t aLength) noexcept
+{
+    if (!m_connection.IsConnected())
+    {
+        return false;
+    }
+
+    // FIXME memory is allocated (and freed) for every packet sent
+    Buffer *pBuffer = GetAllocator()->New<Buffer>(aLength + 8); // Assuming 8 header bytes
+    Buffer::Writer writer(pBuffer);
+    m_connection.WriteHeader(writer, Connection::Header::kPayload);
+    writer.WriteBytes(apData, aLength);
+    bool sent = Send(m_connection.GetRemoteEndpoint(), *pBuffer);
```

Comment:
UDP is limited by an MTU, there needs to be a mechanism for splitting and reassembling if we are over the capacity

Reply:
Ofc I'm aware. I was planning on handling fragmentation in another PR. I haven't thought of a solution I'm satisfied with yet, so it may take a couple of days.